Zonohedra

s214735 opened the pull request:
https://www.notion.so/qim-dtu/Test-Zonohedra-19ebab83b2f980ba8da5dd37392b66de

Implementation of morphological operations with backbone in pygorpho which uses zonohedral kernels for faster computations. Option for scipy when CUDA is not available.

Examples are currently only using scipy. Let me know if I should change/add CUDA examples

delestro reviewed (changes requested):
Thanks for the work on the branch @s214735 !

I think that that there are a few changes that we could make:

1. Let's use only one possible name for each method, to avoid confusion from the user's part. For this, let's keep the most descriptive strings, for example `pygorpho.flat` or `scipy.ndimage` 

2. We need to add a short description of all the methods in the docstrings. The user needs to know the options they can choose from.

3. For the examples, let's keep them as minimal as possible. Here I tested without doing the padding, and it looks fine:
```
s = 8
strel = np.ones((s,s,s))
vol_dilated = qim3d.morphology.dilate(vol, strel, method='flat')
```
4. During the last workshop we decided to keep variable names as descriptive as possible and avoid abbreviations.  So, instead of `strel` let's use `kernel` 

5. Let's make the user's life easier, and accept `kernel` as `int`, `tuple` or `array`. In case it's `int` or `tuple`, we generate the `np.ones()` inside our function

s214735 commented:
No problem!

Removed
s=x
kernel = np.ones(x,x,x)
and changed it to just be in function call for all examples.

Fixed indentation which caused erosion to not be in example box
Additionally noticed that we could make the kernel larger to make the example more clear. I dont think the other examples need it.

+ noted some minor typos and errors along the functions.
